server/sonar-server/src/main/java/org/sonar/server/permission/index/PermissionIndexer.java

```diff
@@ -70,70 +69,60 @@ public PermissionIndexer(DbClient dbClient, EsClient esClient, Collection<Author
     this.dbClient = dbClient;
     this.esClient = esClient;
     this.authorizationScopes = authorizationScopes;
+    this.indexTypes = authorizationScopes.stream()
+      .map(AuthorizationScope::getIndexType)
+      .collect(toSet(authorizationScopes.size()));
   }
 
   @Override
   public Set<IndexType> getIndexTypes() {
-    return authorizationScopes.stream()
-      .map(AuthorizationScope::getIndexType)
-      .collect(toSet(authorizationScopes.size()));
+    return indexTypes;
   }
 
   @Override
   public void indexOnStartup(Set<IndexType> uninitializedIndexTypes) {
+    // TODO do not load everything in memory. Db rows should be scrolled.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)